proxmoxtf/resource_virtual_environment_network.go

```diff
@@ -0,0 +1,373 @@
+	d.Set(mkResourceVirtualEnvironmentNetworkGateway, network.Gateway)
+
+	return nil
+}
+
+func getBody(d *schema.ResourceData, isUpdate bool) *proxmox.VirtualEnvironmentNetworkInterfaceCreateRequestBody {
+	body := &proxmox.VirtualEnvironmentNetworkInterfaceCreateRequestBody{
+		Type: d.Get(mkResourceVirtualEnvironmentNetworkType).(string),
+	}
+	if isUpdate == false {
+		body.Iface = d.Get(mkResourceVirtualEnvironmentNetworkIface).(string)
+	}
+
+	assignIfStringExists(d, &body.Address, mkResourceVirtualEnvironmentNetworkAddress)
+	assignIfStringExists(d, &body.Address6, mkResourceVirtualEnvironmentNetworkAddress6)
+
+	assignIfBoolExists(d, &body.Autostart, mkResourceVirtualEnvironmentNetworkAutostart)
+
+	assignIfStringExists(d, &body.BondPrimary, mkResourceVirtualEnvironmentNetworkBondPrimary)
+	assignIfStringExists(d, &body.BondMode, mkResourceVirtualEnvironmentNetworkBondMode)
+	assignIfStringExists(d, &body.BondXmitHashPolicy, mkResourceVirtualEnvironmentNetworkBondXmitHashPolicy)
+	assignIfStringExists(d, &body.BridgePorts, mkResourceVirtualEnvironmentNetworkBridgePorts)
+
+	assignIfBoolExists(d, &body.BridgeVlanAware, mkResourceVirtualEnvironmentNetworkBridgeVlanAware)
+
+	assignIfStringExists(d, &body.Cidr, mkResourceVirtualEnvironmentNetworkCIDR)
+	assignIfStringExists(d, &body.Cidr6, mkResourceVirtualEnvironmentNetworkCIDR6)
+	assignIfStringExists(d, &body.Comments, mkResourceVirtualEnvironmentNetworkComments)
+	assignIfStringExists(d, &body.Comments6, mkResourceVirtualEnvironmentNetworkComments6)
+	assignIfStringExists(d, &body.Gateway, mkResourceVirtualEnvironmentNetworkGateway)
+	assignIfStringExists(d, &body.Gateway6, mkResourceVirtualEnvironmentNetworkGateway6)
+
+	assignIfIntExists(d, &body.Mtu, mkResourceVirtualEnvironmentNetworkMtu)
+
+	assignIfStringExists(d, &body.NetMask, mkResourceVirtualEnvironmentNetworkNetmask)
+	assignIfStringExists(d, &body.NetMask6, mkResourceVirtualEnvironmentNetworkNetmask6)
+
+	// Ovs Params
+	assignIfStringExists(d, &body.OvsBond, mkResourceVirtualEnvironmentNetworkOvsBond)
+	assignIfStringExists(d, &body.OvsBridge, mkResourceVirtualEnvironmentNetworkOvsBridge)
+	assignIfStringExists(d, &body.OvsOptions, mkResourceVirtualEnvironmentNetworkOvsOptions)
+	assignIfStringExists(d, &body.OvsPorts, mkResourceVirtualEnvironmentNetworkOvsPorts)
+
+	assignIfIntExists(d, &body.OvsTags, mkResourceVirtualEnvironmentNetworkOvsTag)
+	assignIfStringExists(d, &body.OvsSlaves, mkResourceVirtualEnvironmentNetworkOvsPorts)
+
+	// Vlan
+	assignIfIntExists(d, &body.VlanId, mkResourceVirtualEnvironmentNetworkVlanID)
+	assignIfStringExists(d, &body.VlanRawDevice, mkResourceVirtualEnvironmentNetworkVlanRawDevice)
+
+	return body
+}
+
+func assignIfStringExists(d *schema.ResourceData, p **string, key string) {
+	stringVal := d.Get(key).(string)
+
+	if stringVal == "" {
+		return
+	}
+
+	*p = &stringVal
+}
+
+func assignIfBoolExists(d *schema.ResourceData, p **bool, key string) {
```

Comment:
Consider renaming to `resourceVirtualEnvironmentNetworkAssignIfBoolExists` or move it to `utils.go`.